### PR TITLE
add bid prefs to debug tables

### PR DESCRIPTION
**Pull request by rwcarlsen**

### CODE REVIEW

**Comment by mbmcgarry** [comment]
This compiles without error but is reporting preferences of zero even when they should be non-zero.

File: src/exchange_manager.h [hunk 1]
```diff
@@ -95,6 +95,8 @@ class ExchangeManager {
       typename std::set<Bid<T>*>::iterator it4;
       for (it4 = bids.begin(); it4 != bids.end(); ++it4) {
         Bid<T>* b = *it4;
+        Request<T>* r = b->request();
+        double pref = exctx.trader_prefs[b->bidder()][r][b];
```
Comment (gidden) [inline src/exchange_manager.h]: this should probably be keyed on r->requester()

Comment (gidden) [inline src/exchange_manager.h]: speaking of, perhaps access to the `trader_prefs` maps should be read-only..

Comment (rwcarlsen) [inline src/exchange_manager.h]: @mbmcgarry can you confirm if changing ``b->bidder()`` to ``r->requester`` fixes the zero pref issue?

Comment (mbmcgarry) [inline src/exchange_manager.h]: the answer is maybe.  It definitely changed the output in the right direction but not in the way I expected. Which may be an indication of a problem in my code.  . . . I’ll keep investigating.

-Meghan


On Jul 9, 2015, at 9:48 AM, Robert Carlsen <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/exchange_manager.h<https://github.com/cyclus/cyclus/pull/1191#discussion_r34262958>:

> @@ -95,6 +95,8 @@ class ExchangeManager {
>        typename std::set<Bid<T>*>::iterator it4;
>        for (it4 = bids.begin(); it4 != bids.end(); ++it4) {
>          Bid<T>* b = *it4;
> +        Request<T>* r = b->request();
> +        double pref = exctx.trader_prefs[b->bidder()][r][b];


@mbmcgarry<https://github.com/mbmcgarry> can you confirm if changing b->bidder() to r->requester fixes the zero pref issue?

—
Reply to this email directly or view it on GitHub<https://github.com/cyclus/cyclus/pull/1191/files#r34262958>.

*************************************************************************
Meghan McGarry, PhD ~ http://mbmcgarry.github.io/
Postdoctoral Fellow, Consortium for Verification Technology
Computational Nuclear Engineering ~ http://cnerg.engr.wisc.edu
Engineering Physics ~ UW-Madison ~ 608-263-6974
*************************************************************************

Comment (mbmcgarry) [inline src/exchange_manager.h]: Yes!  @gidden 's change makes the preferences record correctly!

**Comment by rwcarlsen** [comment]
conda/binstar issues causing failures :(

**Comment by gidden** [comment]
relaunched

**Comment by gidden** [comment]
looks good, merging

**Pull request merged**